.github/workflows/ci-formatting-and-dependencies.yml

```diff
@@ -25,16 +25,29 @@ jobs:
           black --check .
           npx prettier --check envision/web/src
 
-  check-requirements-change-linux:
+  update-requirements-linux:
     runs-on: ubuntu-18.04
     if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
     container: huaweinoah/smarts:v0.4.18-minimal
     steps:
+      - name: Install git
+        run: |
+          add-apt-repository -y ppa:git-core/ppa && \
+            apt-get update && \
+            apt-get install -y git
       - name: Checkout
         uses: actions/checkout@v2
-      - name: Test pip packages
+      - name: Update requirements
         run: |
           cd $GITHUB_WORKSPACE
-          . ./utils/setup/test_pip_packages.sh
-        continue-on-error: true
-
+          python3.7 -m venv ${{env.venv_dir}}
+          . ${{env.venv_dir}}/bin/activate
+          pip install --upgrade pip
+          pip install .[test,train,camera-obs]
+          pip freeze | grep -v 'smarts' | grep -v 'pkg-resources==0.0.0' > requirements.txt
+      - name: Commit changes
+        uses: EndBug/add-and-commit@v7
+        with:
+          message: 'Update requirements.txt'
+          add: '*.txt'
```

Comment:
Is there a reason we can't just add requirements.txt explicitly here? I'm not sure if it could ever cause a problem, but a wildcard in an automated commit seems slightly dangerous :)

Reply:
Restricted automated commit to `git add requirements.txt`.